website/src/components/canvas/Preview3D.jsx

```diff
@@ -44,15 +63,18 @@ class Preview3D extends React.Component {
   }
 
   /**
-   * If the elements have changed and auto-render is on then update code.
+   * If the elements or canvas orientation have changed and auto-render is on then update code.
    * @param {Object} nextProps  The next props to be recieved by the component.
    * @returns {void}
    */
   componentWillReceiveProps(nextProps) {
     if (nextProps.autoRender) {
-      if (! _.isEqual(this.props.elements, nextProps.elements)) {
-        const newCode = generateScript(nextProps.elements);
-        this.props.dispatch(CodeActions.setCode(newCode));
+      if (! _.isEqual(this.props.elements, nextProps.elements) ||
+          ! _.isEqual(this.props.canvas[CANVAS_ORIENTATION], nextProps.canvas[CANVAS_ORIENTATION])) {
```

Comment:
Don't use ! _.isEqual here for this new line because you are just comparing strings I think, not objects.